### ISSUE TITLE
Undefined class 'UploadFileInfo'. Try changing the name to the name of an existing class, or creating a class with the name 'UploadFileInfo'.

### ISSUE THREAD

**Issue by RudreshNarwal** [flutterchina/dio#501]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by dfmiller** [comment]
UploadFileInfo and FormData.from deprecated in v3.  Documentation needs to be clearer and updated.  I wasted a good part of a day with this.

**Comment by AlexVincent525** [comment]
> UploadFileInfo and FormData.from deprecated in v3. Documentation needs to be clearer and updated. I wasted a good part of a day with this.

Those has been declared loud and clear in the document https://github.com/flutterchina/dio#sending-formdata , so I guess you wasted your time for nothing.

**Comment by moho12n** [comment]
I would like to know what can I use instead of : 
for (var imageFiles in list) { uploadList.add(UploadFileInfo(File(imageFiles), "job.jpg")); }

since UploadFileInfo is no more working

**Comment by AkhilSaji** [comment]
my app gets crashing when i  add dio.


E/Dart    (10356): ../../third_party/dart/runtime/vm/object.cc: 3738: error: Unable to use class Library:'package:dio/src/dio.dart' Class: Dio which is not loaded yet.
E/DartVM  (10356): version=2.7.2 (Mon Mar 23 22:11:27 2020 +0100) on "android_arm64"
E/DartVM  (10356): thread=10387, isolate=main(0x7f75392800)
E/DartVM  (10356):   pc 0x0000007f5ee37d78 fp 0x0000007f5d7b0e30 /data/app/com.cdz.dentcare-2/lib/arm64/libflutter.so+0x1684d78
E/DartVM  (10356): -- End of DumpStackTrace
F/libc    (10356): Fatal signal 6 (SIGABRT), code -6 in tid 10387 (Thread-4)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'xiaomi/mido/mido:7.0/NRD90M/V11.0.2.0.NCFMIXM:user/release-keys'
Revision: '0'
ABI: 'arm64'
pid: 10356, tid: 10387, name: Thread-4  >>> com.cdz.dentcare <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
    #27 pc 0000000001225b5c  /data/app/com.cdz.dentcare-2/lib/arm64/libflutter.so (offset 0x1210000)
    #28 pc 000000000122aa40  /data/app/com.cdz.dentcare-2/lib/arm64/libflutter.so (offset 0x1210000)
    #29 pc 000000000001852c  /system/lib64/libutils.so (_ZN7android6Looper9pollInnerEi+1032)
    #30 pc 0000000000018074  /system/lib64/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+60)
    #31 pc 000000000001110c  /system/lib64/libandroid.so (ALooper_pollOnce+100)
    #32 pc 000000000122a9c4  /data/app/com.cdz.dentcare-2/lib/arm64/libflutter.so (offset 0x1210000)
    #33 pc 0000000001225abc  /data/app/com.cdz.dentcare-2/lib/arm64/libflutter.so (offset 0x1210000)
    #34 pc 00000000012288fc  /data/app/com.cdz.dentcare-2/lib/arm64/libflutter.so (offset 0x1210000)
    #35 pc 00000000000681a4  /system/lib64/libc.so (_ZL15__pthread_startPv+196)
    #36 pc 000000000001db80  /system/lib64/libc

**Comment by kutigbolahan** [comment]
> I would like to know what can I use instead of :
> for (var imageFiles in list) { uploadList.add(UploadFileInfo(File(imageFiles), "job.jpg")); }
> 
> since UploadFileInfo is no more working

please how did you go about it? its giving me issues?

**Comment by horiyahumaid** [comment]
> > I would like to know what can I use instead of :
> > for (var imageFiles in list) { uploadList.add(UploadFileInfo(File(imageFiles), "job.jpg")); }
> > since UploadFileInfo is no more working
> 
> please how did you go about it? its giving me issues?

this is the new class instead of UploadFileInfo:
MultipartFile.fromFile(file.path, filename:fileName)

**Comment by horiyahumaid** [comment]
> I would like to know what can I use instead of :
> for (var imageFiles in list) { uploadList.add(UploadFileInfo(File(imageFiles), "job.jpg")); }
> 
> since UploadFileInfo is no more working

this is the new class instead of UploadFileInfo:
MultipartFile.fromFile(file.path, filename:fileName)
